app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java

```diff
@@ -227,19 +242,28 @@ private void initVisionGlass() {
                 // Note that the system will ask for permissions by itself. This is just a fallback.
                 if (mUSBPermissionRequestCount++ > 1) {
                     showAlertDialog(getString(R.string.phone_ui_usb_alert_dialog_description));
+                    mViewModel.updateConnectionState(PhoneUIViewModel.ConnectionState.PERMISSIONS_UNAVAILABLE);
```

Comment:
Note that I implemented `showAlertDialog`in a way that it calls `finish()` when the user clicks on the OK button, so I guess we need to remove that behaviour as your patch would in theory allow us not to close the app and wait for the user to unplug/plug the glasses, wouldn't it?